### PR TITLE
Cleanup icons, added a few @SideOnlys and cleaned some other stuff up.

### PR DESCRIPTION
**Pull request by iTitus**

### CODE REVIEW

**Comment by BigXplosion** [comment]
this is tested on a server right?

File: src/main/java/buildcraftAdditions/blocks/BlockHeatedFurnace.java [hunk 1]
```diff
@@ -31,14 +31,14 @@
  * http://buildcraftadditions.wordpress.com/wiki/licensing-stuff/
  */
 public class BlockHeatedFurnace extends BlockContainer {
-	public IIcon front, back, sides, top, bottom, frontActivated;
-	public boolean isActivated;
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/blocks/BlockHeatedFurnace.java]: why is this removed?

File: src/main/java/buildcraftAdditions/items/ItemBase.java [hunk 2]
```diff
@@ -19,25 +14,11 @@
  */
 
 public class ItemBase extends Item {
-	public IIcon icon;
-	public String name;
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/items/ItemBase.java]: i don't think this can be removed like that, pretty sure it's used in items extending itemBase or otherwise i've put it there for future use

File: src/main/java/buildcraftAdditions/reference/enums/EnumMachineUpgrades.java [hunk 3]
```diff
@@ -26,17 +24,14 @@
...
-	private EnumMachineUpgrades(String tag, boolean canBeDouble) {
+	private EnumMachineUpgrades(String tag, boolean multipleInstalls) {
 		this.tag = tag;
-		this.multipleInstalls = canBeDouble;
-		if (FMLCommonHandler.instance().getEffectiveSide() == Side.CLIENT) {
-			//DON'T REGISTER ICONS ON THE SERVER SIDE
-			this.texture = new ResourceLocation(Variables.MOD.ID, "textures/items/" + tag + ".png");
-		}
+		this.multipleInstalls = multipleInstalls;
+		texture = new ResourceLocation(Variables.MOD.ID, "textures/items/" + tag + ".png");
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/reference/enums/EnumMachineUpgrades.java]: see the comment you removed, check the side we don't need/want to register the texture at the server side

**Comment by AEnterprise** [comment]
thansk for looking through the entire code, please awnser the line notes and make sure it works on a server and i'll merge it

**Comment by iTitus** [comment]
I tested it extensively, on server and client.

File: src/main/java/buildcraftAdditions/blocks/BlockHeatedFurnace.java (hunk 1 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/blocks/BlockHeatedFurnace.java]: This variable is never used.

File: src/main/java/buildcraftAdditions/blocks/BlockChargingStation.java [hunk 4]
```diff
@@ -32,12 +32,8 @@
...
 public class BlockChargingStation extends BlockContainer {
 
-	public IIcon textureFront;
-	public IIcon textureBack;
-	public IIcon textureLeft;
-	public IIcon textureRight;
-	public IIcon textureTop;
-	public IIcon textureBottom;
+	@SideOnly(Side.CLIENT)
+	private IIcon textureFront, textureBack, textureLeft, textureRight, textureTop, textureBottom;
```
Comment (iTitus) [inline src/main/java/buildcraftAdditions/blocks/BlockChargingStation.java]: textureLeft, textureRight, textureTop are never used

File: src/main/java/buildcraftAdditions/items/ItemBase.java (hunk 2 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/items/ItemBase.java]: You can access itemIcon and if you really want to access the name you can get a substring of the texture name or the unlocalized name.

File: src/main/java/buildcraftAdditions/reference/enums/EnumMachineUpgrades.java (hunk 3 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/reference/enums/EnumMachineUpgrades.java]: Tested and working. Nothing in the class ResourceLocation is @SideOnly and nothing in there loads the texture.
